zmoney/Models/Network/NetworkRequest.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  NetworkRequest.swift
+//  zmoney
+//
+//  Created by Pavel Romanishkin on 12.04.21.
+//
+
+import Foundation
+protocol NetworkRequestType {
+    func sendRequest(
+        to urlString: String,
+        withData data: Data?,
+        withHeaders headers: [String: String]?,
+        using method: String,
+        withCompletion completion: @escaping (Result<Data, Error>) -> Void)
+}
+
+struct NetworkRequest: NetworkRequestType {
+    func sendRequest(
+        to urlString: String,
+        withData data: Data?,
+        withHeaders headers: [String: String]?,
+        using method: String,
+        withCompletion completion: @escaping (Result<Data, Error>) -> Void) {
+        let urlStringEncoded = urlString.addingPercentEncoding(withAllowedCharacters: .urlFragmentAllowed)!
+        guard let url = URL(string: urlStringEncoded) else { return }
+        var request = URLRequest(url: url)
+        request.httpMethod = method
+        if let headers = headers, !headers.isEmpty {
+            for header in headers {
+                request.setValue(header.value, forHTTPHeaderField: header.key)
+            }
+        }
+        request.httpBody = data
+        let session = URLSession.shared
+        session.dataTask(with: request) { (data, response, error) in
+            if let httpResponse = response as? HTTPURLResponse, (400...600).contains(httpResponse.statusCode) {
+                DispatchQueue.main.async {
+                    completion(.failure(NetworkError().returnError(rawValue: httpResponse.statusCode)))
+                }
```

Comment:
Здесь давай `DispatchQueue.main.async` уберем. Это лучше делать на уровне View. Так как в `completion` скорее всего будет некоторая логика, да и вообще, когда запрос в сеть делаем, то ожидаем, что ответ в foreground очереди вернется
```suggestion
                completion(.failure(NetworkError().returnError(rawValue: httpResponse.statusCode)))
```

Reply:
Только когда примешь, то не забудь добавить `DispatchQueue.main.async` на уровне VC